pkg/cmd/ssh/options.go

```diff
@@ -836,24 +837,61 @@ func getNodeNamesFromShoot(f util.Factory, prefix string) ([]string, error) {
 		return nil, errors.New("no Shoot cluster targeted")
 	}
 
-	// create client for the shoot cluster
-	shootClient, err := manager.ShootClient(f.Context(), currentTarget)
+	// create client for the gardener cluster
+	client, err := manager.GardenClient(currentTarget.GardenName())
 	if err != nil {
+		return nil, fmt.Errorf("failed to create garden cluster client: %w", err)
+	}
+
+	// collect names, filter by prefix
+	nodeNames := []string{}
+
+	shoot, err := client.FindShoot(f.Context(), currentTarget.AsListOption())
+	if err != nil {
+		return nil, err
+	}
+
+	newTarget := currentTarget.WithSeedName(*shoot.Spec.SeedName).WithControlPlane(true)
+	// create client for the seed cluster
+	seedClient, err := manager.SeedClient(f.Context(), newTarget)
+	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "forbidden") {
```

Comment:
```go
	if err != nil {
		if !apierrors.IsForbidden(err) {
			return nil, err
		}
		
		shootClient, err := manager.ShootClient(f.Context(), currentTarget)
		if err != nil {
			return nil, err
		}

		// fetch all nodes
		nodes, err := getNodes(f.Context(), shootClient)
		if err != nil {
			return nil, err
		}

		for _, node := range nodes {
			if strings.HasPrefix(node.Name, prefix) {
				nodeNames = append(nodeNames, node.Name)
			}
		}

		return nodeNames, nil
	}
```

use `apierrors.IsForbidden`. Also exit early to avoid deep if-nestings

Reply:
1d6837832a6f2353a4756420079142897404c7bf